gooddata-java/src/main/java/com/gooddata/sdk/service/account/AccountService.java

```diff
@@ -135,17 +135,25 @@ public Account getAccountById(final String id) {
 
     /**
      * Get account for given email
+     * Only domain admin is allowed create new accounts! This means rest request has to authorized as domain admin.     *
      * @param email used as login
      * @param organizationName (domain) in which account is present
-     * @return account for email, empty list if not found
+     * @return account for email
+     * @throws AccountNotFoundException when given account wasn't found
      * @throws GoodDataException when different error occurs
      */
-    public Accounts getAccountByLogin(final String email, final String organizationName) {
+    public Account getAccountByLogin(final String email, final String organizationName) {
         notNull(email, "email");
         notNull(organizationName, "organizationName");
         try {
-            return restTemplate.getForObject(Account.ACCOUNT_BY_EMAIL_URI, Accounts.class, organizationName, email);
-        }  catch (RestClientException e) {
+            final Accounts accounts = restTemplate.getForObject(
+                Account.ACCOUNT_BY_EMAIL_URI, Accounts.class, organizationName, email);
+            if (accounts != null && !accounts.getPageItems().isEmpty()) {
```

Comment:
I would add a check that exactly one account was found. Else throw that multiple results is unexpected.

Reply:
But you're right, that the login should be unique, so it's not necessary ;)